ggp-engine/PhysicsManager.cpp

```diff
@@ -0,0 +1,137 @@
+#include "PhysicsManager.h"
+#include "RigidBody.h"
+#include "Spatial.h"
+
+using namespace DirectX;
+using namespace Physics;
+
+PhysicsManager* PhysicsManager::Instance = nullptr;
+
+const UINT PhysicsManager::GetRigidBodyCount() const
+{
+    return rBodyCount;
+}
+
+const bool Physics::PhysicsManager::Raycast( const DirectX::XMFLOAT3 & aOrigin, const DirectX::XMFLOAT3 & aDirectoin, const float distance )
+{
+    printf( "\tStart raycast!\n" );
+    return false;
+}
+
+PhysicsManager::PhysicsManager()
+{
+    rBodyCount = 0;
+}
+
+PhysicsManager::~PhysicsManager()
+{
+    for ( auto itr = RigidBodyUIDMap.begin(); itr != RigidBodyUIDMap.end(); ++itr )
+    {
+        RigidBody* temp = itr->second;
+        if ( temp != nullptr )
+        {
+            delete temp;
+        }
+    }
+}
+
+PhysicsManager* PhysicsManager::GetInstance()
+{
+    if ( Instance == nullptr )
+    {
+        Instance = new PhysicsManager();
+    }
+    return Instance;
+}
+
+void PhysicsManager::ReleaseInstance()
+{
+    if ( Instance != nullptr )
+    {
+        delete Instance;
+        Instance = nullptr;
+    }
+}
+
+UINT PhysicsManager::AddRigidBody( Spatial* aGameObj, float aMass, EPhysicsLayer aLayer )
+{
+    assert( aGameObj );
```

Comment:
We can use asserts, but so far I've been coding everything to fail without crashing and just output to the console. That way if there is an error, the game will still run and you can see what went wrong.

Reply:
The asserts are removed in release mode, so its really only affecting the debugging / development cycle. We can remove them if you want to though